src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-configuration-page/rgw-configuration-page.component.ts

```diff
@@ -0,0 +1,147 @@
+
+@Component({
+  selector: 'cd-rgw-configuration-page',
+  templateUrl: './rgw-configuration-page.component.html',
+  styleUrls: ['./rgw-configuration-page.component.scss'],
+  providers: [RgwBucketEncryptionModel]
+})
+export class RgwConfigurationPageComponent extends ListWithDetails implements OnInit {
+  readonly vaultAddress = /^((https?:\/\/)|(www.))(?:([a-zA-Z]+)|(\d+\.\d+.\d+.\d+)):\d{4}$/;
+
+  kmsProviders: string[];
+
+  columns: Array<CdTableColumn> = [];
+
+  configForm: CdFormGroup;
+  permissions: Permissions;
+  encryptionConfigValues: any = [];
+  selection: CdTableSelection = new CdTableSelection();
+
+  @Output()
+  submitAction = new EventEmitter();
+  authMethods: string[];
+  secretEngines: string[];
+  tableActions: CdTableAction[];
+  bsModalRef: NgbModalRef;
+  filteredEncryptionConfigValues: {};
+  excludeProps: any[] = [];
+  disableCreate = false;
+  allEncryptionValues: any;
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwBucketService: RgwBucketService,
+    public authStorageService: AuthStorageService,
+    private rgwEncryptionModal: RgwBucketEncryptionModel,
+    private modalService: ModalService
+  ) {
+    super();
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`Encryption Type`,
+        prop: 'encryption_type',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Key Management Service Provider`,
+        prop: 'backend',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Address`,
+        prop: 'addr',
+        flexGrow: 1
+      }
+    ];
+    const addAction: CdTableAction = {
+      permission: 'create',
+      icon: Icons.add,
+      name: this.actionLabels.CREATE,
+      click: () => this.openRgwConfigModal(false),
+      disable: () => this.disableCreateAction()
+    };
+    const editAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      name: this.actionLabels.EDIT,
+      click: () => this.openRgwConfigModal(true)
+    };
+    this.tableActions = [addAction, editAction];
+    this.rgwBucketService.getEncryptionConfig().subscribe((data: any) => {
+      this.allEncryptionValues = data;
+      const allowedBackends = this.rgwEncryptionModal.kmsProviders;
+      const kmsBackends = new Set();
+      data['SSE-KMS'].forEach((item: { backend: unknown }) => kmsBackends.add(item.backend));
+      const s3Backends = new Set();
+      data['SSE-S3'].forEach((item: { backend: unknown }) => s3Backends.add(item.backend));
+      const allKmsBackendsPresent = allowedBackends.every((backend) => kmsBackends.has(backend));
+      const allS3BackendsPresent = allowedBackends.every((backend) => s3Backends.has(backend));
+      if (allKmsBackendsPresent && allS3BackendsPresent) {
+        this.disableCreate = true;
+      } else {
+        this.disableCreate = false;
+      }
```

Comment:
feels to me like they can add more types of backends and we'll end up adding more loops here to iterate through them and disable the create button. this should be atleast somewhat simplified for future developers.

```
// get the backend set
getBackend(encryptionData, encryptionType) {
  return new Set(encryptionData[encryptionType].map((item) => item.backend));
}

// check if the allowed backends are present.
areAllAllowedBackendsPresent(allowedBackends, backendsSet) {
  return allowedBackends.every((backend) => backendsSet.has(backend));
}

const allowedBackends = this.rgwEncryptionModal.kmsProviders;

const kmsBackends = getBackend(data, 'SSE-KMS');
const s3Backends = getBackend(data, 'SSE-S3');

const allKmsBackendsPresent = areAllAllowedBackendsPresent(allowedBackends, kmsBackends);
const allS3BackendsPresent = areAllAllowedBackendsPresent(allowedBackends, s3Backends);

this.disableCreate = allKmsBackendsPresent && allS3BackendsPresent;
```